tests/functional/object/mcg/test_s3_vector.py

```
@@ -1310,3 +1310,558 @@ def test_vector_distance_metrics(self, vector_bucket_factory, mcg_obj):
+            returnDistance=True,
+        ).get("vectors", [])
+        assert len(cat_a_results) > 0, "Expected at least one category='A' result"
+        for result in cat_a_results:
+            result_meta = result.get("metadata", {})
+            assert result_meta.get("category") == "A", (
+                f"Non-category='A' vector in filtered results: key={result.get('key')}, "
+                f"metadata={result_meta}"
+            )
+        logger.info(
+            f"✓ All {len(cat_a_results)} results have category='A' — "
+            f"metadata filter correctly excludes category='B' and 'A_updated' vectors"
```

Ruff (0.15.17)

[warning] Logging statement uses f-string

(G004)